src/test/util/matchable.rs

```diff
@@ -67,7 +67,7 @@ impl Matchable for Document {
                     return false;
                 }
             } else {
-                return false;
+                return v == &Bson::Null;
```

Comment:
The spec test for checking that `allowDiskUse` is not sent as part of a find when it's not specified as an option [explicitly includes an `allowDiskUse` field in the CommandStartedEvent expectation with a null value](https://github.com/mongodb/specifications/blob/7077aa39aafe1abd6e5d5529b13d66ab9ded90bf/source/crud/tests/v2/find-allowdiskuse.json#L20-L30). Given that we prefer to sync the spec tests as-is instead of modifying them, the most straightforward fix in code is to have our matching functionality treat a null value in the expected document as a match for a missing value in our actual document.

Reply:
Should this only return when `v != &Bson::Null`? Otherwise I think this is terminating the loop too early.

Reply:
Oops, yes, you're right. Fixed!

Reply:
I think we're still returning too early here actually, since we still need to inspect the other keys. The `if` should have the `v == &Bson::Null` in it and the body still just have `return false` I think. I believe we have the same bug in the Swift driver test runner actually: https://github.com/mongodb/mongo-swift-driver/blob/main/Tests/MongoSwiftSyncTests/SpecTestRunner/Match.swift#L75

Reply:
The latest version of the code has the following code:

```rust
if let Some(actual_v) = self.get(k) {
    if !actual_v.matches(v) {
        return false;
    }
} else if v != &Bson::Null {
    return false;
}            
```

This will return false if there is a value for the key in the actual document (i.e. in `self`) and it doesn't match the expected key, and it will return false if the key doesn't in the actual document and the value in the actual document isn't null (meaning the value that the test expects is not present in the actual document). Otherwise, it doesn't return and continues looping to check the other keys. This seems correct to me; can you clarify what types of keys and values in expected/actual will cause it to terminate too early?

Reply:
Ah I think I was looking at an outdated diff, it looks correct to me.